### DISCUSSION TITLE
hierarchically sort selction based on syntax tree

### DISCUSSION THREAD

**Discussion by ArvidHammarlund** [helix-editor/helix#10481] [Q&A] [answered]
Is it possible to sort text hierarchically rather than by line? What would be the easiest way to split the selection based on syntax level so that i can call the sort command? 

Eg. I want to make the following nix code...
```
{
     foo = {
         foo = 1;
         bar = 2;
         baz = 3;
     };
     bar = 1;
     baz = 2;
}
```
...become this..
```
{
     bar = 1;
     baz = 2;
     foo = {
         bar = 2;
         baz = 3;
         foo = 1;
     };
}
```

**Comment by pascalkuthe** [comment, marked as answer]
That is not supported and probably won't be in core. You would need to do multiple sorts for each scope. On latest master you can use `A-I` to select each assignment in the scope that contains the cursor and then cursor those
